clippy_lints/src/utils/hir_utils.rs

```diff
@@ -635,24 +580,15 @@ impl<'a, 'tcx: 'a> SpanlessHash<'a, 'tcx> {
     pub fn hash_stmt(&mut self, b: &Stmt) {
         match b.node {
             StmtKind::Local(ref local) => {
-                let c: fn(_) -> _ = StmtKind::Local;
```

Comment:
By removing this, the discriminant needs to be hashed at the beginning of the function. I'll add this.